src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -310,5 +312,63 @@ public IVsPathContext GetSolutionPathContext()
                 pathContext.FallbackPackageFolders.Cast<string>(),
                 trie);
         }
+
+        private IList<Project> GetProjectsInSolution(DTE dte)
+        {
+            var supportedProjects = dte.Solution.Projects;
+            Projects projects = dte.Solution.Projects;
+
+            List<Project> projectList = new List<Project>();
+            var item = projects.GetEnumerator();
+            while (item.MoveNext())
+            {
+                var project = item.Current as Project;
+                if (project == null)
+                {
+                    continue;
+                }
+
+                if (project.Kind == ProjectKinds.vsProjectKindSolutionFolder)
+                {
+                    IEnumerable<Project> solutionFolderProjects = GetSolutionFolderProjects(project);
+                    projectList.AddRange(solutionFolderProjects);
+                }
+                else
+                {
+                    projectList.Add(project);
+                }
+            }
+
+            return projectList;
+        }
+
+        /// <summary>
+        /// Gets all projects that are not solution folders within the given solutionFolder.
+        /// (i.e. If the hierarchy is src\a\b.xproj, src\c.xproj; the resulting list will contain b.xproj and c.xproj).
+        /// </summary>
+        private IEnumerable<Project> GetSolutionFolderProjects(Project solutionFolder)
+        {
+            List<Project> projectList = new List<Project>();
```

Comment:
See rule 10 in https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#c-coding-style.